internal/controller/dnsrecord_controller.go

```diff
@@ -202,14 +222,54 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return r.updateStatusAndRequeue(ctx, previous, dnsRecord, time.Second)
 	}
 
-	if !controllerutil.ContainsFinalizer(dnsRecord, DNSRecordFinalizer) {
-		logger.Info("Adding Finalizer", "finalizer_name", DNSRecordFinalizer)
-		controllerutil.AddFinalizer(dnsRecord, DNSRecordFinalizer)
-		err = r.Update(ctx, dnsRecord)
-		if err != nil {
-			return ctrl.Result{}, err
+	if r.IsLocalRecord() {
+		if !controllerutil.ContainsFinalizer(dnsRecord, DNSRecordFinalizer) {
+			logger.Info("Adding Finalizer", "finalizer_name", DNSRecordFinalizer)
+			controllerutil.AddFinalizer(dnsRecord, DNSRecordFinalizer)
+			err = r.Update(ctx, dnsRecord)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{RequeueAfter: randomizedValidationRequeue}, nil
 		}
-		return ctrl.Result{RequeueAfter: randomizedValidationRequeue}, nil
+
+		if probesEnabled {
+			if err = r.ReconcileHealthChecks(ctx, dnsRecord, allowInsecureCert); err != nil {
+				return ctrl.Result{}, err
+			}
+			// get all probes owned by this record
+			if err := r.List(ctx, probes, &client.ListOptions{
+				LabelSelector: labels.SelectorFromSet(map[string]string{
+					ProbeOwnerLabel: BuildOwnerLabelValue(dnsRecord),
+				}),
+				Namespace: dnsRecord.Namespace,
+			}); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+
+		if r.IsSecondary() && dnsRecord.IsDelegating() {
```

Comment:
@philbrookes I think this should just happen for either role (primary|secondary) when delegating for it to work in a primary-primary setup. There is probably to consider in this kind of setup though as we only really want one of them writing the status, or maybe we should be limiting what we write to the status of a remote record entirely.

Reply:
yes, that sounds like a good change to me.